### DISCUSSION TITLE
release-1.1 milestone

### DISCUSSION THREAD

**Discussion by CecileRobertMichon** [kubernetes-sigs/cluster-api-provider-azure#1818] [Ideas]
What should we work on for the next month? Now that v1beta1 / v1.0.0 is behind us (almost), let's start thinking about what we should tackle next. We will pick the top 3-5 ideas at the next community office hours to focus on in the release-1.1 milestone, with the aim to release on December 15th. Please share links to issues below and upvote the others that you would like to see in the next release!

Note: let's keep this discussion thread limited to issue links only, and discuss each issue individually in the issue's comment thread if needed.

**Comment by CecileRobertMichon** [comment]
Add AzureClusterTemplate type so we can support ClusterClass: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1500

**Comment by CecileRobertMichon** [comment]
Enable async reconciliation across more/all services: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1702

**Comment by shysank** [comment]
Migrate templates to use ClusterClass: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1826. (Optimistic)

**Reply by mboersma** [reply]
This is dependent on CAPI finalizing ClusterClass support.

**Comment by shysank** [comment]
Consistent logging #1806

**Comment by devigned** [comment]
Complete remaining work to add features and move AzureManagedClusters out exp.

**Reply by devigned** [reply]
Here are some things I'd like to see:
- refactoring to using the reconciler / spec structure other reconcilers use so we can clean up some of the service / reconciler code
- add conditions and provide better status during reconcile loops
- have better initial status, like we have started to reconcile, but are not quite ready yet. I think this is coincides with making the reconcilers async
- make the reconcilers async to provide better feedback to users
- tinker with the watches and reconcilers to ensure we are squeezing as quick of reconciliation as possible

**Reply by alexeldeib** [reply]
https://github.com/kubernetes-sigs/cluster-api/issues/4526 // https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1503 are also critical imo

**Comment by devigned** [comment]
Support Azure CNI see #467
